src/pages/waiting-room/cat-d/waiting-room.cat-d.page.ts

```diff
@@ -0,0 +1,252 @@
+import { Component, ViewChild, OnInit } from '@angular/core';
+import { FormGroup } from '@angular/forms';
+import { IonicPage, NavController, NavParams, Platform, Navbar, ModalController } from 'ionic-angular';
+import { AuthenticationProvider } from '../../../providers/authentication/authentication';
+import { Store, select } from '@ngrx/store';
+import { StoreModel } from '../../../shared/models/store.model';
+import * as waitingRoomActions from '../waiting-room.actions';
+import { Observable } from 'rxjs/Observable';
+import {
+  getPreTestDeclarations,
+} from '../../../modules/tests/pre-test-declarations/common/pre-test-declarations.reducer';
+import * as preTestDeclarationsActions
+  from '../../../modules/tests/pre-test-declarations/common/pre-test-declarations.actions';
+import {
+  getInsuranceDeclarationStatus,
+  getResidencyDeclarationStatus,
+  getSignatureStatus,
+} from '../../../modules/tests/pre-test-declarations/common/pre-test-declarations.selector';
+import { getCandidate } from '../../../modules/tests/journal-data/cat-c/candidate/candidate.cat-c.reducer';
```

Comment:
How are tackling instances where we have to reference other categories for now? Sticking with the TODO approach or something different?

Reply:
i guess TODO approach.  But working through the sub tasks for ticket 4291 we these issues will be resolved one by one